torch/hub.py

```diff
@@ -645,7 +658,14 @@ def _legacy_zip_load(filename, model_dir, map_location):
     return torch.load(extracted_file, map_location=map_location)
 
 
-def load_state_dict_from_url(url, model_dir=None, map_location=None, progress=True, check_hash=False, file_name=None):
+def load_state_dict_from_url(
+    url: str,
+    model_dir: Optional[str] = None,
+    map_location: Optional[Union[Callable[[str], str], Dict[str, str]]] = None,
+    progress: bool = True,
+    check_hash: bool = False,
+    file_name: Optional[str] = None
+) -> nn.Module:
```

Comment:
@adamjstewart @jbschlosser The return annotation is wrong. It should be `Dict[str, Any]` keeping it aligned with `nn.Module.state_dict()`

https://github.com/pytorch/pytorch/blob/b30c027abf6560dafb88d67ebd446851a5729651/torch/nn/modules/module.py#L1308

This leads to [CI failures](https://github.com/pytorch/pytorch/blob/b30c027abf6560dafb88d67ebd446851a5729651/torch/nn/modules/module.py#L1308) in `torchvision`.

Reply:
Should we revert or forward fix?

Reply:
I'll submit a PR to fix this, shouldn't take more than a second

Reply:
Nvm, just saw #76895